### PR TITLE
Tessa/remove old components

### PR DESCRIPTION
**Pull request by tesk9**
```
This is a MAJOR change.

---- REMOVED MODULES - MAJOR ----

    Nri.Ui.Checkbox.V4
    Nri.Ui.DisclosureIndicator.V1
    Nri.Ui.Dropdown.V2
    Nri.Ui.PremiumCheckbox.V1
    Nri.Ui.PremiumCheckbox.V2
    Nri.Ui.PremiumCheckbox.V3
    Nri.Ui.SegmentedControl.V7
    Nri.Ui.Select.V6
    Nri.Ui.SlideModal.V1
    Nri.Ui.TextArea.V3
    Nri.Ui.TextInput.V3
    Nri.Ui.TextInput.V4
    Nri.Ui.TextInput.V5


---- Nri.Ui.Html.Attributes.V2 - MINOR ----

    Added:
        targetBlank : List.List (Html.Styled.Attribute msg)


---- Nri.Ui.Html.V3 - MAJOR ----

    Removed:
        type alias Options = { preventDefault : Bool, stopPropagation : Bool }
        nbsp : Html msg
        onEnter : a -> Attribute a
        onEnterAndSpace : msg -> Attribute msg
        onEsc : a -> a -> Attribute a
        oxfordifyWithHtml :
            String -> String -> List (Html msg) -> List (Html msg)
        role : String -> Attribute msg
        textFromList : List String -> Html msg

    Changed:
      - defaultOptions : Options
      + defaultOptions :
            { preventDefault : Basics.Bool, stopPropagation : Basics.Bool }

      - onKeyUp : Options -> (Int -> Maybe a) -> Attribute a
      + onKeyUp :
            { preventDefault : Basics.Bool, stopPropagation : Basics.Bool }
            -> (Basics.Int -> Maybe.Maybe a)
            -> Html.Styled.Attribute a

```

### CODE REVIEW

**Comment by avh4** [comment]
@tesk9 would you be okay waiting a few more days on this? If so, then I think Callout, Alert, and AlertBanner will be able to be removed too (details https://noredink.slack.com/archives/C7CQZM7CL/p1588016092002400 )

**Comment by tesk9** [comment]
Yup, for sure!

**Comment by avh4** [comment]
Just merged https://github.com/NoRedInk/noredink-ui/pull/502.  Hopefully I can remove NoRedInk uses of Alert and AlertBanner by tomorrow.

**Comment by avh4** [comment]
https://github.com/NoRedInk/NoRedInk/pull/27371 merged, so Alert.V4 and AlertBanner.V6 can be removed now (unless it gets revered in Monday's master QA)

File: Makefile [hunk 1]
```diff
@@ -40,7 +40,9 @@ checks:
 
 .PHONY: diff
 diff: node_modules
-	if (npx elm diff | tee /dev/stderr | grep -q MAJOR); then echo "MAJOR changes are not allowed!"; exit 1; fi
+	true
+# 	Temporarily disabled to allow for an intentional MAJOR upgrade
+# 	if (npx elm diff | tee /dev/stderr | grep -q MAJOR); then echo "MAJOR changes are not allowed!"; exit 1; fi
```
Comment (juanedi) [inline Makefile:45]: ❓  Does this mean we need to release this snapshot of the code and then make another commit reverting this change to the Makefile?

Comment (tesk9) [reply Makefile:45]: Yes!
